### PR TITLE
feat: add bot notification base template of function hosted trigger

### PR DESCRIPTION
**Pull request by a1exwang**
add function base template (without trigger files)

template source code is based on dev branch of this repo: https://github.com/qinezh/mybotapp/tree/dev

https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/13776366/

### CODE REVIEW

**Comment by a1exwang** [comment]
E2E TEST: n/a

File: templates/bot/ts/notification-function-base/src/sdk/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { Activity, Middleware, TurnContext } from "botbuilder";
+import { ConversationReferenceStore } from "./store";
+
+export interface NotificationMiddlewareOptions {
+  conversationReferenceStore: ConversationReferenceStore;
+}
+
+enum ActivityType {
```
Comment (yukun-dong) [inline templates/bot/ts/notification-function-base/src/sdk/middleware.ts:8]: With the discussions about enum and literal types in previous [PR](https://github.com/OfficeDev/TeamsFx/pull/4152) , I wonder if we can update it to be like `Unknown = 'unknown'`.

Comment (a1exwang) [reply templates/bot/ts/notification-function-base/src/sdk/middleware.ts:8]: Yes. The SDK folder is currently just to make the scaffold code compiles fine. Later they will be implemented in teamsfx-sdk. So they will be removed here. Can we improve that in sdk later?

File: templates/bot/ts/notification-function-base/src/internal/initialize.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// try better naming to hide for user
```
[the remark this answers is not in the document]
Comment (a1exwang) [reply templates/bot/ts/notification-function-base/src/internal/initialize.ts]: Removed

**Review by yukun-dong — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Review by XiaofuHuang — APPROVED** [review]

**Pull request merged**